signature/verifier.go

```diff
@@ -39,32 +39,46 @@ func NewVerifierFromFiles(certPaths []string) (*Verifier, error) {
 	return &Verifier{TrustedCerts: certs}, nil
 }
 
+func ValidatePayloadContentType(payload *signature.Payload) error {
```

Comment:
> Should we also validate that the given payload corresponds to the specified contentType?

https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#payload
`payload.ContentType` can only be `application/vnd.cncf.notary.payload.v1+json`.  What do I need to do to validate the payload?

Reply:
you can open an issue for this since it wasn't there before refactoring, ideally we want to make sure that `application/vnd.cncf.notary.payload.v1+json` content type abides by following rules
```
targetArtifact : Required property whose value is the descriptor of the target artifact manifest that is being signed. Both [OCI descriptor](https://github.com/opencontainers/image-spec/blob/main/descriptor.md) and [ORAS artifact descriptors](https://github.com/oras-project/artifacts-spec/blob/main/descriptor.md) are supported.
        Descriptor MUST contain mediaType, digest, size fields.
        Descriptor MAY contain annotations and if present it MUST follow the [annotation rules](https://github.com/opencontainers/image-spec/blob/main/annotations.md#rules). Notary v2 uses annotations for storing both Notary specific and user defined metadata. The prefix io.cncf.notary in annotation keys is reserved for use in Notary v2 and MUST NOT be used outside this specification.
        Descriptor MAY contain artifactType field for artifact manifests, or the config.mediaType for oci.image based manifests.
```

Reply:
Creating an issue to track this problem https://github.com/notaryproject/notation-go/issues/81